src/SignalRServiceExtension/TriggerBindings/Resolver/SignalRRequestResolver.cs

```diff
@@ -24,10 +35,31 @@ public bool ValidateContentType(HttpRequestMessage request)
             return contentType == Constants.JsonContentType || contentType == Constants.MessagePackContentType;
         }
 
+        // The algorithm is defined in spec: Hex_encoded(HMAC_SHA256(access-key, connection-id))
         public bool ValidateSignature(HttpRequestMessage request, string accessToken)
         {
-            //TODO: Add real signature validation
-            return true;
+            if (!_validateSignature)
+            {
+                return true;
+            }
+
+            if (!string.IsNullOrEmpty(accessToken) &&
+                request.Headers.TryGetValues(Constants.AsrsSignature, out var values))
+            {
+                var signatures = SignalRTriggerUtils.GetSignatureList(values.FirstOrDefault());
+                if (signatures == null)
+                {
+                    return false;
+                }
+                using (var hmac = new HMACSHA256(Encoding.UTF8.GetBytes(accessToken)))
```

Comment:
maybe move to the protocol project which will be shared by both side?

Reply:
As it's a common HMAC algorithm, and there're existing implementation for each language. I don't put it in protocol for now.